### ISSUE TITLE
Intermittent issue with asb/etcd not coming up from run_latest_build.sh  

### ISSUE THREAD

**Issue by jwmatthews** [openshift/ansible-service-broker#299]
https://github.com/openshift/ansible-service-broker/blob/master/scripts/run_latest_build.sh

$ oc get pods
NAME                    READY     STATUS             RESTARTS   AGE
asb-2357364550-krm3l    0/1       CrashLoopBackOff   2          49s
etcd-2338997634-jk8jv   0/1       CrashLoopBackOff   2          49s

$ oc logs asb-2357364550-krm3l
Using config file mounted to /etc/ansible-service-broker/config.yaml
============================================================
==           Starting Ansible Service Broker...           ==
============================================================
[2017-07-24T20:03:47.136Z] [NOTICE] Initializing clients...
[2017-07-24T20:03:47.136Z] [DEBUG] Trying to connect to etcd
[2017-07-24T20:03:47.136Z] [INFO] == ETCD CX ==
[2017-07-24T20:03:47.136Z] [INFO] EtcdHost: etcd
[2017-07-24T20:03:47.136Z] [INFO] EtcdPort: 2379
[2017-07-24T20:03:47.136Z] [INFO] Endpoints: [http://etcd:2379]
[2017-07-24T20:03:48.137Z] [ERROR] client: etcd cluster is unavailable or misconfigured; error #0: client: endpoint http://etcd:2379 exceeded header timeout

$ oc logs etcd-2338997634-jk8jv
2017-07-24 20:03:11.733160 W | pkg/flags: unrecognized environment variable ETCD_PORT_2379_TCP_PORT=2379
2017-07-24 20:03:11.733427 W | pkg/flags: unrecognized environment variable ETCD_SERVICE_PORT=2379
2017-07-24 20:03:11.733434 W | pkg/flags: unrecognized environment variable ETCD_PORT_2379_TCP_ADDR=172.30.147.171
2017-07-24 20:03:11.733451 W | pkg/flags: unrecognized environment variable ETCD_SERVICE_HOST=172.30.147.171
2017-07-24 20:03:11.733458 W | pkg/flags: unrecognized environment variable ETCD_PORT=tcp://172.30.147.171:2379
2017-07-24 20:03:11.733462 W | pkg/flags: unrecognized environment variable ETCD_PORT_2379_TCP=tcp://172.30.147.171:2379
2017-07-24 20:03:11.733467 W | pkg/flags: unrecognized environment variable ETCD_PORT_2379_TCP_PROTO=tcp
2017-07-24 20:03:11.733472 W | pkg/flags: unrecognized environment variable ETCD_SERVICE_PORT_ETCD_ADVERTISE=2379
2017-07-24 20:03:11.733493 I | etcdmain: etcd Version: 3.2.4
2017-07-24 20:03:11.733506 I | etcdmain: Git SHA: c31bec0
2017-07-24 20:03:11.733511 I | etcdmain: Go Version: go1.8.3
2017-07-24 20:03:11.733515 I | etcdmain: Go OS/Arch: linux/amd64
2017-07-24 20:03:11.733520 I | etcdmain: setting maximum number of CPUs to 16, total number of available CPUs is 16
2017-07-24 20:03:11.733959 N | etcdmain: the server is already initialized as member before, starting as etcd member...
2017-07-24 20:03:11.735000 I | embed: listening for peers on http://localhost:2380
2017-07-24 20:03:11.735072 I | embed: listening for client requests on 0.0.0.0:2379
2017-07-24 20:03:11.736553 C | etcdserver: create snapshot directory error: mkdir /data/member/snap: permission denied

$ oc get pv | grep Bound
pv0073    100Gi      RWO,ROX,RWX   Recycle         Bound       ansible-service-broker/etcd                            3m

$ oc describe pv pv0073
Name:		pv0073
Labels:		volume=pv0073
Annotations:	pv.kubernetes.io/bound-by-controller=yes
StorageClass:	
Status:		Bound
Claim:		ansible-service-broker/etcd
Reclaim Policy:	Recycle
Access Modes:	RWO,ROX,RWX
Capacity:	100Gi
Message:	
Source:
    Type:	HostPath (bare host directory volume)
    Path:	/var/lib/origin/openshift.local.pv/pv0073
Events:		<none>

$ ls -larth /var/lib/origin/openshift.local.pv/pv0073
ls: cannot open directory '/var/lib/origin/openshift.local.pv/pv0073': Permission denied
jmatthews@beast r (master) $ sudo ls -larth /var/lib/origin/openshift.local.pv/pv0073
total 12K
drwxr-xr-x. 103 root       root 4.0K Mar 15 12:29 ..
drwx------.   4 1000170000 root 4.0K Jul 21 10:01 member
drwxrwx---.   3 root       root 4.0K Jul 24 16:05 .

**Comment by jmontleon** [comment]
I was seeing this error:

```
"[2017-07-24T20:03:48.137Z] [ERROR] client: etcd cluster is unavailable or misconfigured; error #0: client: endpoint http://etcd:2379 exceeded header timeout"
```

Which I believe was occurring because I was missing docker rules on the FORWARDING chain in iptables. Restarting firewalld I saw a set of new rules at the top of my FORWARD chain and was able to deploy properly.

iptables -nvL FORWARD
```
 4706 4726K DOCKER-ISOLATION  all  --  *      *       0.0.0.0/0            0.0.0.0/0           
 4414 4675K DOCKER     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0           
 4385 4673K ACCEPT     all  --  *      docker0  0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED
  292 50919 ACCEPT     all  --  docker0 !docker0  0.0.0.0/0            0.0.0.0/0           
   29  1740 ACCEPT     all  --  docker0 docker0  0.0.0.0/0            0.0.0.0/0           
    0     0 ACCEPT     all  --  *      *       0.0.0.0/0            0.0.0.0/0            ctstate RELATED,ESTABLISHED

```

This was on a new host where docker and firewalld were started in rapid succession. It's possible there was a race condition that caused this issue, since from what I can see when (re)starting firewalld without docker started excludes these rules, while restarting firewalld with docker started includes them. 

Long story short, this may be due to an environment issue rather than a problem with openshift or the broker.

**Comment by jwmatthews** [comment]
@jmontleon so looks like this issue can manifest in 2 different ways:

1. Reusing PV shares with stale data, led to etcd showing this error: 
| etcdserver: create snapshot directory error: mkdir /data/member/snap: permission denied
Once I added a clean in the scripts to wipe out prior location for PV shares, the problem went away for me. 

2. Firewalls issues as you saw.

**Comment by rthallisey** [comment]
I haven't seen this in a while. We can re open if it's still an issue.
